cpp/src/arrow/record_batch.cc

```diff
@@ -393,7 +393,7 @@ Result<std::shared_ptr<RecordBatchReader>> RecordBatchReader::Make(
 RecordBatchReader::~RecordBatchReader() {
   auto st = this->Close();
   if (!st.ok()) {
-    ARROW_LOG(WARNING) << "Implicitly called RecordBatchReader::Close failed: " << st;
+    st.Warn("Implicitly called RecordBatchReader::Close failed: ");
```

Comment:
Sorry for the confusion, I have clicked the worng line. I meant this line. WDYT?